### PR TITLE
Add SlashCommand.guild_ids

### PR DESCRIPTION
**Pull request by Dorukyum**
## Summary
Also refactored discord/commands.py

## Checklist
- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/commands.py [hunk 1]
```diff
@@ -42,45 +39,51 @@ def __new__(cls, *args, **kwargs):
...
         return self
+
     def __init__(self, func, *args, **kwargs):
         if not asyncio.iscoroutinefunction(func):
-            raise TypeError('Callback must be a coroutine.')
+            raise TypeError("Callback must be a coroutine.")
+        self.callback = func
 
-        name = kwargs.get('name') or func.__name__
+        self.guild_ids = kwargs.get("guild_ids", None)
```
Comment (Middledot) [inline discord/commands.py:48]: Should it be defaulted to list?

Comment (Dorukyum) [reply discord/commands.py:48]: It should be defaulted to `None` to make the command global

File: discord/commands.py [hunk 2]
```diff
@@ -42,45 +39,51 @@ def __new__(cls, *args, **kwargs):
...
 
-        description = kwargs.get('description') or inspect.cleandoc(func.__doc__)
-        if description == None:
-            description = "No description set"
-        elif not isinstance(name, str):
-            raise TypeError('Description of a command must be a string.')
+        description = (
+            kwargs.get("description")
+            or inspect.cleandoc(func.__doc__)
+            or "No description set"
```
Comment (BobDotCom) [inline discord/commands.py:58]: We should raise an error if no description is given. it makes more sense as an extension of the API

File: discord/commands.py [hunk 3]
```diff
@@ -42,45 +39,51 @@ def __new__(cls, *args, **kwargs):
...
+        )
+        if not isinstance(name, str):
+            raise TypeError("Description of a command must be a string.")
         self.description = description
 
         options = OrderedDict(inspect.signature(func).parameters)
         options.pop(list(options)[0])
         for a, o in options:
-            if o.name == None:
+            if not o.name:
```
Comment (BobDotCom) [inline discord/commands.py:67]: if o.name is None would be better if we are only checking for None

File: discord/commands.py [hunk 4]
```diff
@@ -42,45 +39,51 @@ def __new__(cls, *args, **kwargs):
...
-    
     def to_dict(self):
         return {
-            "name":self.name,
-            "description":self.description,
-            "options":[o.to_dict() for o in self.options]
+            "name": self.name,
+            "description": self.description,
+            "options": [o.to_dict() for o in self.options],
+            "guild_ids": self.guild_ids,
```
Comment (BobDotCom) [inline discord/commands.py:76]: We want guild_ids to be left out of this if guild_ids is None

**Review by BobDotCom — CHANGES_REQUESTED** [review]

**Review by BobDotCom — APPROVED** [review]

**Pull request merged**